main.go

```diff
@@ -99,19 +106,57 @@ func sendRequest(
 	response, err := client.Do(req)
 
 	if err != nil {
-		received <- &MeasuredResponse{0, 0, 0, false, err}
+		received <- &MeasuredResponse{0, 0, 0, false, false, err}
 	} else {
-		if sz, err := io.CopyBuffer(ioutil.Discard, response.Body, bodyBuffer); err == nil {
-			response.Body.Close()
-
-			received <- &MeasuredResponse{
-				uint64(sz),
-				response.StatusCode,
-				elapsed.Nanoseconds() / 1000000,
-				false,
-				nil}
+		if !checkHash {
+			if sz, err := io.CopyBuffer(ioutil.Discard, response.Body, bodyBuffer); err == nil {
+				response.Body.Close()
+
+				received <- &MeasuredResponse{
+					uint64(sz),
+					response.StatusCode,
+					elapsed.Nanoseconds() / 1000000,
+					false,
+					false,
+					nil}
+			} else {
+				received <- &MeasuredResponse{0, 0, 0, false, false, err}
+			}
 		} else {
-			received <- &MeasuredResponse{0, 0, 0, false, err}
+			//var measure MeasuredResponse
+			var buf bytes.Buffer
+			var writer io.Writer
+			usedGzip := false
+			if response.Uncompressed {
+				writer = &buf
```

Comment:
I couldn't get this to work in the Uncompressed case. I have a go server that returns "hello\n". When I set `-hashSampleRate 1 -hashValue 12230792304413059251 -compress`, then I get 100% matches, but when I omit the compressed flag, then I get 100% mismatches. I couldn't figure out what the issue is.